libarchive/archive_read_support_format_iso9660.c

```diff
@@ -951,6 +979,51 @@ isPVD(struct iso9660 *iso9660, const unsigned char *h)
 	return (48);
 }
 
+static int
+isRootDirectoryRecord(const unsigned char *p) {
+	int flags;
+
+	/* ECMA119/ISO9660 requires that the root directory record be _exactly_ 34 bytes.
+	 * However, we've seen images that have root directory records up to 68 bytes. */
+	if (p[DR_length_offset] < 34 || p[DR_length_offset] > 68) {
+		return (0);
+	}
+
+	/* The root directory location must be a 7.3.3 32-bit integer. */
```

Comment:
Again, a lot of these integers are stored twice, so we can verify that the two copies have the same value.